tests/net/socket/tcp/src/main.c

```diff
@@ -234,15 +234,59 @@ void test_v6_send_recv(void)
 	k_sleep(TCP_TEARDOWN_TIMEOUT);
 }
 
-void test_v4_sendto_recvfrom(void)
+/* Test the stack behavior with a resonable sized block data, be sure to have multiple packets */
+#define TEST_BLOCK_SIZE 60000
+
+static uint8_t testBlockTcp[TEST_BLOCK_SIZE];
+static uint8_t testBlockTcpRx[TEST_BLOCK_SIZE + 10];
+
+#define TCP_SERVER_STACK_SIZE 4096
+
+K_THREAD_STACK_DEFINE(tcp_server_stack_area, TCP_SERVER_STACK_SIZE);
+struct k_thread tcp_server_thread_data;
+
+/* A thread that receives, while the other part transmits */
+void tcp_server_block_thread(void* vps_sock, void* unused2, void* unused3) 
+{
+	int new_sock;
+	struct sockaddr addr;
+	int* ps_sock = (int*)vps_sock;
+	socklen_t addrlen = sizeof(addr);
+
+	test_accept(*ps_sock, &new_sock, &addr, &addrlen);
+	zassert_equal(addrlen, sizeof(struct sockaddr_in), "wrong addrlen");
+
+	/* check the received data */
+	ssize_t recved = 0;
+	ssize_t total_received = 0;
+	int iteration = 0;
+	while (total_received < TEST_BLOCK_SIZE) {
+		recved = recv(new_sock, testBlockTcpRx + total_received, sizeof(testBlockTcpRx) - total_received, 0);
+		
+		zassert(recved > 0, "received bigger then 0",
+			"Error receiving bytes %i bytes, got %i on top of %i in iteration %i, errno %i", TEST_BLOCK_SIZE,
+			recved, total_received, iteration, errno);
+		total_received += recved;
+		iteration++;
+	}
+	zassert_equal(memcmp(testBlockTcpRx, testBlockTcp, sizeof(testBlockTcp)),
+		0,
+		"unexpected data");
+	
+	test_close(new_sock);
+}
+
+void test_v4_send_recv_block(void)
```

Comment:
I wonder about the name here, `block` suggests that the test is about the blocking socket, while it's not essential here. Maybe rename it to `test_v4_send_recv_large`?

Reply:
Good suggestion, done